src/js/components/App.js

```diff
@@ -18,14 +18,31 @@ type Props = {
   selectedGenre: string,
   view: View,
   posterData: PosterData,
+  match?: { params: {id: number} },
+  location?: {search: URL },
   onChangeView: () => {},
-  onFilmSelect: () => {}
+  onFilmSelect: (id: number) => {}
 };
 
 export class App extends Component<Props, {}> {
 
   componentDidMount() {
-    this.props.fetchData(`${helpers.routes.base}/movies`);
+    const params: any | {id?: number} = this.props.match &&
+      this.props.match.params ? this.props.match : false;
```

Comment:
why do we check for existence of `this.props.match.params` if we assign `this.props.match` ?